autofocus/predict/process__predict_example.R

```diff
@@ -0,0 +1,230 @@
+#Examples of how to make requests agains the image classification endpoints
```

Comment:
Is the double underscore in the filename intentional?

Reply:
Nope, that is definitely a typo

Reply:
Regarding the model probabilities. That is something that I was unaware of until you brought it up today (that they were independent). I'm thinking that if we are trying to select  the 'best' one it may make the most sense to divide each probability by the sum of all the probabilities. 

For example, if we have
```
{
raccoon:0.80,
rabbit:0.20,
coyote:0.75
}
```

Then we would divide each of those element by `0.80+0.20+0.75`. That would at least ensure that the relative probabilities sum to 1. On top of this, an image with multiple 'high' probabilities for different classifications (as in the above example) would get down-weighted a bit while those with a single 'high' probability would be penalized less.

Finally, I do agree that this could be made MUCH more modular and a library would be a great way to do that.

Reply:
> Then we would divide each of those element by 0.80+0.20+0.75. That would at least ensure that the relative probabilities sum to 1.

Why would we want that? The idea behind letting them be independent is that the categories are not mutually exclusive, so we should not force them to sum to 1.

Reply:
While that is true in this specific case, what do you do with an image that is `{'raccoon': 0.99, 'coyote': 0.99}. In our decade of camera trapping we've only gotten one image of a coyote and a raccoon so assuming that both are in the is a little suspect. Aside from human and dog the likelihood of getting two unique species is quite low. However, if we allow the probabilities to sum to one you could just add together the human and dog probabilities (making a human AND dog classifier). At the end of the day though these types of summaries can be done after the fact (post autofocus) and we can then make some comparisons about which way performs better.

Reply:
> While that is true in this specific case, what do you do with an image that is {'raccoon': 0.99, 'coyote': 0.99}?

If the model gives that result and the image doesn't contain both a raccoon and a coyote, then the model got it wrong. Cases where it is that badly wrong are extremely rare.

> However, if we allow the probabilities to sum to one you could just add together the human and dog probabilities (making a human AND dog classifier).

If the app returns `{"human": 1, "dog": 1}` (with zeros for other categories), that means that the model is confident that the image contains *both* a human *and* a dog. If it returns `{"human": .5, "dog": .5}` (with zeros for other categories), that means that the model is maximally uncertain about whether the image contains a human and about whether it contains a dog. This is a very important distinction, and we would lost it if we made the numbers sum to one after the fact.

If we trained a multiclass rather than multilabel model so that `{"human": 1, "dog": 1}` was impossible, then it is hard to say what the model would do on an image that contains both humans and dogs. For instance, in that case `{"human": .5, "dog": .5}` could mean that the image contains both humans and dogs but the model doesn't have the means to represent that fact, or it could mean that the model is confident that the image contains something but not whether that thing is a human or a dog.

The categories are not mutually exclusive, so treating them as separate labels is the right approach in principle. We can revisit the approach if we find that it doesn't work well in practice, but so far I don't see any reason to think that it wouldn't.

Reply:
Let's meet up or jump on a call if what I'm saying isn't clear or if it is clear but you disagree.